cfpb/sbl-frontend
# Type of financial institution (Phase 1.5)

Closes #455 

## Changes

- Integrates Type of FI form into the workflow
- Updates Type of FI form content
- Prevents creation of a Filing for the institution until they've clicked "Start filing"


## How to test this PR

1. Create an institution without any sbl types (Example JSON below)
2. Associate LEI with the domain (local example)
   - POST `{{regtech_url}}/v1/institutions/{{lei}}/domains`
   - body:
    ```
    [
      {
        "domain": "local.host"
      }
    ]
    ```
3. Click on `Start filing` for that institution (Card heading should read "Start the filing process (institution)", which is a temporary adjustment to help identify if the user will be routed to Types of FI `(institution)` or to the Upload page `(filing)`)
4. Should be directed to `/institutions/{your-lei}/type/2024`
5. You can fill out the form (not required to `Save and continue`)
6. Click `Save and continue`
7. Should be directed to an intermediary page that creates a Filing for that institution (`Loading filing data...`)
7. Once Filing is created, should be auto redirected to the Upload page for that institution


## Screenshots
### Flow for institution without existing SBL types
https://github.com/cfpb/sbl-frontend/assets/2592907/e8f3550e-1574-4ec1-a7f4-716b1b3f2fef

### FILING HOME: Institution that DOES NOT already have SBL types (`institution`) (will be routed to Type of FI form)
<img width="699" alt="Screenshot 2024-05-09 at 10 55 17 AM" src="https://github.com/cfpb/sbl-frontend/assets/2592907/bfd4af6a-f524-420b-b16f-7799ad0a9e54">

### FILING HOME: Institution that already has SBL types (`filing`) but has not yet started the filing process
<img width="702" alt="Screenshot 2024-05-09 at 10 55 10 AM" src="https://github.com/cfpb/sbl-frontend/assets/2592907/d8dd6d22-b5da-4e91-8932-ee0356282f0f">

### Type of FI form
![screencapture-localhost-8899-institution-123456789MEISBANK808-type-2024-2024-05-09-10_59_31](https://github.com/cfpb/sbl-frontend/assets/2592907/d35c80a9-7c97-44ab-a48a-2cbb4075f6c6)




## Institution without sbl types
```
{
  "name": "Meis Bank 808",
  "lei": "123456789MEISBANK808",
  "is_active": true,
  "tax_id": "88-9999991",
  "rssd_id": 9991,
  "primary_federal_regulator_id": "OCC",
  "hmda_institution_type_id": "1",
  "sbl_institution_types": [],
  "hq_address_street_1": "Test Address Street 1",
  "hq_address_street_2": "",
  "hq_address_city": "Test City 1",
  "hq_address_state_code": "GA",
  "hq_address_zip": "00000",
  "parent_lei": "PARENTTESTBANK999",
  "parent_legal_name": "PARENT TEST BANK 999",
  "parent_rssd_id": 99945,
  "top_holder_lei": "TOPHOLDERLEI999",
  "top_holder_legal_name": "TOP HOLDER LEI 999",
  "top_holder_rssd_id": 999456
}
```
